dvc/ignore.py

```diff
@@ -148,6 +148,37 @@ def __eq__(self, other):
         return self.basenames == other.basenames
 
 
+class DvcWarningDirs:
+    special_dirs = {
+        ".petest_cache",
+        ".mypy_cache",
+        "node_modules",
+        ".venv",
+        "venv",
+        "env",
+        "__pycache__",
+        "dist",
+        "eggs",
+        "wheels",
+        "htmlcov",
+        "docs",
+        ".ipynb_checkpoints",
+        ".vscode",
+        ".github",
+    }
```

Comment:
Hm, makes me wonder if we should really just ignore them by default (like we do with .git and .hg), and if some user really needs these, he could `!something` in dvcignore to include it.

Or, we might consider generating a default .dvcignore (kinda like github generates .gitignore when you create a project) to make it extra explicit. Though this is simple on `init`, but managing this stuff in existing projects is not too simple :slightly_frowning_face: 

I'm just worried about these warnings getting too annoying. Plus, unless I'm missing something, ignore execution order is not deterministic (that's actually is a bigger problem in general for us), so you never know if this filter will be executed before or after .dvcignores in which user has ignored a special dir explicitly. :slightly_frowning_face:

Reply:
One more thing that this makes me think about is having some kind of safe guard, that would, for example, record the amount of time it took to walk through the directory and warn if it didn't find anything but it took, say, more than 5 sec.

Reply:
This issue was opened after this conversation https://github.com/iterative/dvc/pull/3257. It was decided that it is better to warn the user if a special directory is found than to ignore it by default. But annoyance from warnings can be a problem. Maybe we should make it possible to disable this warning? From the other hand, the default .dvcignore also good idea because the ignore will be obvious to the user.

Reply:
What makes idea of default `.dvcignore` compelling, is the fact that is not only explicit, but also editable by user.

If we decide to go with warnings, they definitely need turn off mechanism, just like we did with `slow_link_guard`.

Reply:
Now I soppouse that the default ``` .dvcignore```, is better option. It`s more explicite and easy to configure. Also the list of special dirs always will be incomplite, we can not add all posiible variants here. And default ``` .dvcignore``` provide additional example for users how to deal with such cases.

 @efiop, @pared, if you are agree with me, then I will change the code. Should I close this PR and open a new one later, or commit right here?

Reply:
👍 I love the idea of an explicit `.dvcignore`

Reply:
I wonder whether it should be default. Maybe we should add some option to init, or dedicate special command for that? It seems to me that `.dvcignore` was supposed to let user decide what is part of his project. If we always create default `.dvcignore` on init, we make it our choice. @efiop @aerubanov, what do you think?

Reply:
I think that default ```.dvcignore``` should consist comment with link to the dvcignore documentation. And we should add option for init to filling ```.dvcignore``` with some content. Not all list of special folder, rather based on type of project specified by user.  Something like github gitignore template https://github.com/github/gitignore, but with fewer options ofcorse.

Reply:
@aerubanov I think you are right on this one.